src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -927,7 +927,16 @@ Namespace Microsoft.CodeAnalysis.Semantics
         Private Function CreateBoundBlockOperation(boundBlock As BoundBlock) As IBlockStatement
             Dim statements As Lazy(Of ImmutableArray(Of IOperation)) = New Lazy(Of ImmutableArray(Of IOperation))(
                 Function()
-                    Return boundBlock.Statements.Select(Function(n) Create(n)).Where(Function(s) s.Kind <> OperationKind.None).ToImmutableArray()
+                    Return boundBlock.Statements.Select(Function(n) Create(n)).Where(
+                        Function(statement)
+                            ' Don't include the method/operator/accessor declaration in the block
+                            Return statement.Syntax.Kind() <> SyntaxKind.OperatorStatement AndAlso
+                                   statement.Syntax.Kind() <> SyntaxKind.SubStatement AndAlso
+                                   statement.Syntax.Kind() <> SyntaxKind.FunctionStatement AndAlso
+                                   statement.Syntax.Kind() <> SyntaxKind.GetAccessorStatement AndAlso
+                                   statement.Syntax.Kind() <> SyntaxKind.SetAccessorStatement
+                        End Function
+                    ).ToImmutableArray()
```

Comment:
`SubNewStatement` already has a test. I'll add tests for the 3 event blocks as well.